packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -509,4 +495,130 @@ export class QuoteRequestor {
         const results = (await Promise.all(quotePromises)).filter(x => x !== undefined);
         return results as Array<RfqQuote<ResponseT>>;
     }
+    private async _fetchAndValidateFirmQuotes(
+        makerToken: string,
+        takerToken: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        comparisonPrice: BigNumber | undefined,
+        options: RfqRequestOpts,
+        assetOfferings: RfqMakerAssetOfferings,
+    ): Promise<SignedNativeOrder[]> {
+        const quotesRaw = await this._getQuotesAsync<V4RFQFirmQuote>(
+            makerToken,
+            takerToken,
+            assetFillAmount,
+            marketOperation,
+            comparisonPrice,
+            options,
+            'firm',
+            assetOfferings,
+        );
+        const quotes = quotesRaw.map(result => ({ ...result, response: result.response.signedOrder }));
+
+        // validate
```

Comment:
nit: I would go even further and have a non-async validate quotes function, but not necessary